u3-views/features/videoController.js

```diff
@@ -0,0 +1,19 @@
+const { getVideo, getSuggestions } = require('../_services/fakeapi')
+
+const videoDetails = (req, res) => {
+  const videoId = req.params.id
+  const video = getVideo(videoId)
+  const teases = getSuggestions(6)
+  res.render('video-player', { video: video, teases: teases })
+}
+const videoHome = (req, res) => {
+  const videoId = req.query.id
+  const video = getVideo(videoId)
+  const teases = getSuggestions(30)
+  res.render('video-home', { video: video, teases: teases })
```

Comment:
เส้นทางนี้คือ `/videos` ไม่มีตัวแปรเส้นทาง `req.query.id` จะไม่มีอยู่จริง คุณควรลบ `videoId` และ `video`:
This route is `/videos`. It doesn't have any route variables. `req.query.id` will never exist. You should delete `videoId` and `video`:
```diff
-  const videoId = req.query.id
-  const video = getVideo(videoId)
  const teases = getSuggestions(30)
-  res.render('video-home', { video: video, teases: teases })
+  res.render('video-home', { teases: teases })
```

Reply:
Thanks for fixing this! Much better!